lib/vector-buffers/src/buffer_usage_data.rs

```diff
@@ -313,3 +351,36 @@ impl BufferUsage {
         spawn_named(task.instrument(span.or_current()), task_name.as_str());
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::thread;
+
+    #[test]
+    fn test_multithreaded_updates_are_correct() {
+        const NUM_THREADS: u64 = 16;
+        const INCREMENTS_PER_THREAD: u64 = 10_000;
+
+        let counter = Arc::new(AtomicU64::new(0));
+
+        let mut handles = vec![];
+
+        for _ in 0..NUM_THREADS {
+            let counter = Arc::clone(&counter);
+            let handle = thread::spawn(move || {
+                for _ in 0..INCREMENTS_PER_THREAD {
+                    update_counter(&counter, 1);
+                    update_counter(&counter, -1);
+                }
+            });
+            handles.push(handle);
+        }
+
+        for handle in handles {
+            handle.join().unwrap();
+        }
+
+        assert_eq!(counter.load(Ordering::Relaxed), 0);
+    }
```

Comment:
suggestion add test to check negative value: 
```
#[test]
fn test_update_counter_prevents_negatives() {
  let counter = AtomicU64::new(100);
  
  update_counter(&counter, -50);
  assert_eq!(counter.load(Ordering::Relaxed), 50);
        
  update_counter(&counter, -100);
  assert_eq!(counter.load(Ordering::Relaxed), 0);

  update_counter(&counter, -50);
  assert_eq!(counter.load(Ordering::Relaxed), 0);
}
```

Reply:
Added this and an overflow test in b840c9e2a